File: leaderboard/api/web/Models/Team.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for details.
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace Sentinel.Models
{
    public class Team
    {
        [Key]
        public String Id {get;set;}
        public string TeamName { get; set; }
        public int DownTimeMinutes { get; set; }
        public int Points { get; set; }
        public bool IsScoringEnabled { get; set; }

        [JsonProperty(NullValueHandling=NullValueHandling.Ignore)]
        public List<ServiceStatus> ServiceStatus {get;set;}

        public Team()
        {
            Id = Guid.NewGuid().ToString();
        }
    }
}

File: leaderboard/web/src/app/pages/teams/team.ts
import { ITeam } from "../../shared/team";
import { IServiceStatus } from "../../shared/servicestatus";

export class Team implements ITeam{

    public teamName: string;
    public serviceStatus: IServiceStatus[];
    public id: string;

    public downTimeMinutes: number;
    public points: number;
    public isScoringEnabled: boolean;
    constructor (
    ){
        this.downTimeMinutes = 0;
        this.points = 0;
        this.isScoringEnabled = false;
        this.teamName = 'Enter Team Name'
        this.serviceStatus = null;
    }

}
